core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1556,12 +1728,141 @@ public void warmUp() {
+                if (colType != detColType) {
+                    if (colType == -detColType) { // a column was deleted
+                        if (colType < detColType) {
+                            detachedDeleteExtraColNames.add(columnName);
+                        } else {
+                            // column was added to the table, and does not exist in attaching
+                            detachedAddMissingColNames.add(columnName);
+                        }
+                        colsNegativeTypeMatch = true;
+                    } else {
+                        throw CairoException.detachedColumnMetadataMismatch(colIdx, columnName, "type");
+                    }
+                }
+
+                // check is indexed
+                if (metadata.isColumnIndexed(colIdx) != detachedMetadata.isColumnIndexed(detColIdx)) {
+                    throw CairoException.detachedColumnMetadataMismatch(colIdx, columnName, "is_indexed");
+                }
+            }
+            // detached could have had more columns than we need
+            for (int i = columnCount, limit = detachedMetadata.getColumnCount(); i < limit; i++) {
+                int detColIdx = detachedMetadata.getWriterIndex(i);
+                String columnName = detachedMetadata.getColumnName(detColIdx);
+                detachedDeleteExtraColNames.add(columnName);
+            }
+
+            // check structure version
+            if (colsNegativeTypeMatch) {
+                if (metadata.getStructureVersion() == detachedMetadata.getStructureVersion()) {
+                    throw CairoException.detachedMetadataMismatch("structure_version should be different");
+                }
+            } else if (metadata.getStructureVersion() != detachedMetadata.getStructureVersion()) {
+                throw CairoException.detachedMetadataMismatch("structure_version");
+            }
+
+            // load/check _cv, updating local column tops
+            // set current _cv to where the partition was
+            other2.concat(COLUMN_VERSION_FILE_NAME).$();
+            if (!ff.exists(other2)) {
+                // Backups and older versions of detached partitions will not have _cv
+                LOG.info().$("detached _cv not found, skipping check [path=").$(other2).I$();
+                return;
+            }
+
+            if (detachedColumnVersionReader == null) {
+                detachedColumnVersionReader = new ColumnVersionReader();
+            }
+            detachedColumnVersionReader.ofRO(ff, other2);
+            detachedColumnVersionReader.readSafe(MillisecondClockImpl.INSTANCE, Long.MAX_VALUE);
+            LongList entries = columnVersionWriter.getCachedList();
+            LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+            for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
+                // detachedEntries: [partitionTimestamp, columnIndex, columnNameTxn, columnTop]*
```

Comment:
This comment probably mean the structure of Column Version record. Can you incapsulate this loop inside ColumnVersionWriter instead of hardcoding indexes / record length here?